### PR TITLE
ref: misc. improvements around "stage" concept

### PR DESCRIPTION
**Pull request by jorgeorpinel**
Per https://github.com/iterative/dvc.org/pull/3223#pullrequestreview-865256066

### CODE REVIEW

File: content/docs/command-reference/index.md [hunk 1]
```diff
@@ -10,16 +10,14 @@ does not change directories in your terminal).
 
 ## Typical DVC workflow
 
-- In an existing Git repository, initialize a <abbr>DVC project</abbr> with
-  `dvc init`.
-- Copy data files or dataset directories for modeling into the repository, and
-  track them with DVC using the `dvc add` command.
-- Process the data with your own source code, using `dvc.yaml` and/or the
-  `dvc stage add` command to specify further <abbr>outputs</abbr> that should
```
Comment (jorgeorpinel) [inline content/docs/command-reference/index.md:18]: Ended up rewriting this whole section 😅

File: content/docs/command-reference/freeze.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # freeze
 
-Freeze [stages](/doc/command-reference/stage) until `dvc unfreeze` is used on
+Freeze [stages](/doc/command-reference/run) until `dvc unfreeze` is used on
```
Comment (jorgeorpinel) [inline content/docs/command-reference/freeze.md:3]: These are for consistency (only place where `stage` ref is linked insted of `run`) for now. Wil be properly addressed in #2883.

File: content/docs/command-reference/plots/modify.md [hunk 3]
```diff
@@ -134,8 +134,8 @@ plots:
 
 ## Example: Template change
 
-_dvc stage add --plots file.csv ..._ command assign the default template that
-needs to be changed in many cases. A simple command changes the template:
+Something like `dvc stage add --plots file.csv ...` assigns the default
```
Comment (jorgeorpinel) [inline content/docs/command-reference/plots/modify.md:137]: We can auto-link to https://dvc.org/doc/command-reference/stage/add#--plots directly from here now.

**Review by jorgeorpinel — COMMENTED** [review]
Explanation of some changes 👇

File: content/docs/command-reference/import-url.md [hunk 4]
```diff
@@ -100,9 +100,12 @@ DVC supports several types of external locations (protocols):
   necessary to track if the specified URL changed.
 
 Another way to understand the `dvc import-url` command is as a shortcut for
-generating a pipeline stage with and external dependency. This is discussed in
-the [External Dependencies](/doc/user-guide/external-dependencies)
-documentation, where an alternative is demonstrated for each of these schemes.
+generating a pipeline [stage](/doc/command-reference/run) with and external
```
Comment (dberenbaum) [inline content/docs/command-reference/import-url.md:103]:
```suggestion
generating a pipeline [stage](/doc/command-reference/run) with an external
```

**Review by dberenbaum — COMMENTED** [review]
One type, otherwise looks good!

**Review by dberenbaum — APPROVED** [review]

**Pull request merged**